sql/analyzer/indexes.go

```diff
@@ -341,53 +285,67 @@ func getComparisonIndexLookup(
 		left, right, e = swapTermsOfExpression(e)
 	}
 
-	if !isEvaluable(left) && isEvaluable(right) {
-		gf := expression.ExtractGetField(left)
-		if gf == nil {
-			return nil, nil
-		}
+	if isEvaluable(left) || !isEvaluable(right) {
+		return nil, nil
+	}
```

Comment:
doing gods work here, we have way too many giga-blocks

Reply:
💪